pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -584,9 +513,12 @@ func ResolvePipelineTask(
 		PipelineTask: &pipelineTask,
 	}
 	rpt.CustomTask = rpt.PipelineTask.TaskRef.IsCustomTask() || rpt.PipelineTask.TaskSpec.IsCustomTask()
-	switch {
-	case rpt.IsCustomTask() && rpt.PipelineTask.IsMatrixed():
-		rpt.RunObjectNames = getNamesOfRuns(pipelineRun.Status.ChildReferences, pipelineTask.Name, pipelineRun.Name, pipelineTask.Matrix.CountCombinations())
+	numberOfTaskRuns := 1
```

Comment:
I suggest renaming this to `numCombinations`. The number of TaskRuns can change during execution, but the number of combinations should not.